lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -109,37 +109,51 @@ using namespace Js;
                 return IndexType_PropertyId;
             }
         }
-        else
+
+        if (JavascriptNumber::Is(indexVar))
```

Comment:
>Is [](start = 30, length = 2)

should you use Is_NoTaggedIntCheck here?

Reply:
Yep we can use it in this case